src/common/user-store/preferences-helpers.ts

```diff
@@ -222,6 +237,31 @@ const syncKubeconfigEntries: PreferenceDescription<KubeconfigSyncEntry[], Map<st
   },
 };
 
+const editorConfiguration: PreferenceDescription<EditorConfiguration, EditorConfiguration> = {
+  fromStore(val) {
+    const result: EditorConfiguration = {};
+
+    result.lineNumbers = val?.lineNumbers ?? defaultEditorConfig.lineNumbers;
+    result.tabSize = val?.tabSize ?? defaultEditorConfig.tabSize;
+    result.miniMap = val?.miniMap ?? defaultEditorConfig.miniMap;
+
+    return result;
+  },
```

Comment:
We can use `merge` utility here to save all props at once https://lodash.com/docs/4.17.15#merge